### PR TITLE
fix(logout): make logout update state after session is invalidated

### LINKED ISSUE #153: Logout button does not signal when logged out for functional tests
This: https://github.com/mozilla/123done/blob/oauth/static/js/123done.js#L62

```
      // upon logout, make an api request to tear the user's session down
      $.post('/api/logout');
```

The post ajax request above provides no feedback, so functional tests cannot tell when the session was invalidated. 

Causes: https://github.com/mozilla/fxa-content-server/issues/3721

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #153

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson @vbudhram r?

File: static/js/123done.js [hunk 1]
```diff
@@ -43,23 +43,29 @@ $(document).ready(function() {
...
+          // clear items from the dom at logout
+          $("#todolist > li").remove();
+          State.save();
+
+          // don't display the warning icon at logout time, but wait until the user
+          // makes a change to her tasks
+          $("#dataState > div").css('display', 'none');
+        })
+        .fail(function() {
+          // this should ever happen
```
Comment (shane-tomlinson) [inline static/js/123done.js]: `ever` => `never`

**Comment by shane-tomlinson** [comment]
This solution makes total sense, given the problems listed in the content server bug. Thanks @vladikoff and @jbuck for digging in. I pushed a commit to fix the typo, as soon as the tests go green, merge it!

File: static/js/123done.js (hunk 1 above)
Comment (vladikoff) [inline static/js/123done.js]: never ever!

**Pull request merged**